### ISSUE TITLE
bigquery: Support math/big.Int as parameter type

### ISSUE THREAD

**Issue by Thomasdezeeuw** [googleapis/google-cloud-go#4703]
I'm using [QueryParameter](https://pkg.go.dev/cloud.google.com/go/bigquery#QueryParameter) to send parameters along with my query. On the BigQuery side I'm using `BIGNUMERIC(32)` (no precision, i.e. an integer) as type. Normally you would use `int64` as type, however my input doesn't fit in 64 bits. I tried using `big.Rat` but BigQuery doesn't allow silent truncation (and rightfully so).

Go has nice package for large numbers: `math/big`, which is already partially support through `big.Rat`. However `big.Int` doesn't seem to be supported. Is it possible for this client to support `big.Int`? I'm will to give it a go if it's an acceptable change (and someone can point into the right direction w.r.t. the code).

**Comment by shollyman** [comment]
I'll need to look at this.  The issue is we try to determine the bigquery type from the go type.  Clearly we need to consider another mechanism for the user to explicitly provide the destination BQ type.

**Comment by Thomasdezeeuw** [comment]
@shollyman maybe this should be another issue, but I would also be happy to provide/receive the value as a string as send to/returned by BigQuery (as I assume underneath it's all HTTP anyway, right?). I know it doesn't really fit the current API, but could that be a (future) possibility?

**Comment by alvarowolfx** [comment]
@Thomasdezeeuw recently we merged some work to add support for explicit typed query parameters. Although right now you can't pass a `big.Int` directly, you can convert it to a string and explicitly inform the type. Here is an example:

```
ctx := context.Background()
client, err := bigquery.NewClient(ctx, "project-id")
if err != nil {
	// TODO: Handle error.
}
bigIntParam := big.NewInt(1234567890123456789)
q := client.Query("select name, price from product where price > @bigint")
q.Parameters = []bigquery.QueryParameter{
	{
		Name: "bigint", 
		Value: &QueryParameterValue{
			Type: StandardSQLDataType{
				TypeKind: "BIGNUMERIC",
			},
			Value: bigIntParam.String(),
		},
	},
}
```

Do you think is solves your issue ?

**Comment by Thomasdezeeuw** [comment]
@alvarowolfx yes that works. We just switched to that API for https://github.com/googleapis/google-cloud-go/issues/4704, I forgot to mention it here.

I'm ok with closing this issue.
